main.c

```diff
@@ -684,6 +746,55 @@ static int slirp4netns_config_from_options(struct slirp4netns_config *cfg,
     cfg->disable_host_loopback = opt->disable_host_loopback;
     cfg->enable_sandbox = opt->enable_sandbox;
     cfg->enable_seccomp = opt->enable_seccomp;
+    if (opt->outbound_addr != NULL) {
+#if SLIRP_CONFIG_VERSION_MAX >= 2
+        if (inet_pton(AF_INET, opt->outbound_addr,
+                      &cfg->outbound_addr.sin_addr) == 1) {
+            cfg->enable_outbound_addr = true;
+            cfg->outbound_addr.sin_family = AF_INET;
+            cfg->outbound_addr.sin_port = 0; // Any local port will do
+        } else {
+            fprintf(stderr, "outbound-addr has to be valid ipv4 address.");
```

Comment:
Can we also support specifying the address by the eth dev name (`SIOCGIFADDR`)?

Reply:
Something like this: https://stackoverflow.com/a/21028879? 
Would you like to introduce for that `--outbound-interface` or expose it through existing flags. 

What about portability? I do not know much about ioctl calls portability. I wonder whether `SIOCGIFADDR` would not limit us somehow.

Note: for ipv6 we will need `SIOCGIFADDR_IN6`.

Reply:
What do you mean by portability? slirp4netns is Linux-only.

Reply:
I meant portability across different distributions.

Reply:
Why isn't it portable?

Reply:
Heh I didn't say it is not. I was just concerned and wanted to double check with you if it is ok to use it.

Reply:
Actually I used getifaddrs. I did not find a proper way how to use ioctl for obtaining ipv6, but getifaddrs works for both. `getifaddrs` was introduced in glibc 2.3.3. I assume that should be alright as it was released in 2003.